### PR TITLE
Integration of ServiceMesh

### PR DESCRIPTION
**Pull request by VaishnaviHire**
This commit includes adding ServiceMesh opertor alongwith the prerequisites.

This doesn't include setup for ServiceMesh like creating control plane and member roll.

**Related PR**:

https://github.com/opendatahub-io/manifests/pull/77

### CODE REVIEW

File: openshift-service-mesh/control-plane.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        install:
+          storage:
+            type: Memory
+      kiali:
+        enabled: true
+      prometheus:
+        enabled: true
+    version: v2.0
+    telemetry:
+      type: Istiod
```
Comment (crobby) [inline openshift-service-mesh/control-plane.yaml:25]: newline

File: openshift-service-mesh/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - subscriptions
+  - namespace.yaml
+  - memberroll.yaml
+  - controlplane.yaml
```
Comment (crobby) [inline openshift-service-mesh/kustomization.yaml:9]: newline

File: openshift-service-mesh/memberroll.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: maistra.io/v1
+  kind: ServiceMeshMemberRoll
+  metadata:
+    namespace: istio-system
+    name: default
+  spec:
+    members:
+      - kubeflow
```
Comment (crobby) [inline openshift-service-mesh/memberroll.yaml:8]: newline

**Comment by crobby** [comment]
@VaishnaviHire I'm pretty sure this needs a rebase to hopefully get the tests working again.  Then I'm hoping we can get this merged

**Comment by VaishnaviHire** [comment]
> @VaishnaviHire I'm pretty sure this needs a rebase to hopefully get the tests working again. Then I'm hoping we can get this merged

Rebased

**Comment by crobby** [comment]
> > @VaishnaviHire I'm pretty sure this needs a rebase to hopefully get the tests working again. Then I'm hoping we can get this merged
> 
> Rebased. Also should this be merged after the manifest PR [opendatahub-io/manifests#77](https://github.com/opendatahub-io/manifests/pull/77) ?

That's a good question.  I'm hoping to see this get merged soon so that I can use it for some work on kserve.  Who do we need to pester for reviews on these PRs?

**Comment by VaishnaviHire** [comment]
> > > @VaishnaviHire I'm pretty sure this needs a rebase to hopefully get the tests working again. Then I'm hoping we can get this merged
> > 
> > 
> > Rebased. Also should this be merged after the manifest PR [opendatahub-io/manifests#77](https://github.com/opendatahub-io/manifests/pull/77) ?
> 
> That's a good question. I'm hoping to see this get merged soon so that I can use it for some work on kserve. Who do we need to pester for reviews on these PRs?

Anyone who would like to try out ServiceMesh with Kubeflow applications. I know @LaVLaS was planning on reviewing the PRs. I am also happy to help with testing steps, setting up environment.

**Review by LaVLaS — APPROVED** [review]
/lgtm

This worked for me when providing ServiceMesh for the KF 1.4 on OCP deployment outlined in https://github.com/opendatahub-io/manifests/pull/81

**Pull request merged**